mmselfsup/models/algorithms/mae_classification.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from ..builder import ALGORITHMS, build_backbone, build_head
+from ..utils import Mixup
+from .base import BaseModel
+
+
+@ALGORITHMS.register_module()
+class MAEClassification(BaseModel):
+    """Simple image classification for MAE classification benchmark.
+
+    Args:
+        backbone (dict): Config dict for module of backbone.
+        with_sobel (bool): Whether to apply a Sobel filter.
+            Defaults to False.
+        head (dict): Config dict for module of loss functions.
+            Defaults to None.
+    """
+
+    def __init__(self,
+                 backbone,
+                 head=None,
+                 init_cfg=None,
+                 mixup_alpha=None,
+                 cutmix_alpha=None,
+                 cutmix_minmax=None,
+                 prob=None,
+                 switch_prob=None,
+                 mode=None,
+                 label_smoothing=None,
+                 num_classes=None,
+                 finetune=True):
+        super(MAEClassification, self).__init__(init_cfg)
+        self.backbone = build_backbone(backbone)
+        assert head is not None
+        self.head = build_head(head)
+        self.finetune = finetune
+        if self.finetune:
+            self.mix_up = Mixup(
+                mixup_alpha=mixup_alpha,
+                cutmix_alpha=cutmix_alpha,
+                cutmix_minmax=cutmix_minmax,
+                prob=prob,
+                switch_prob=switch_prob,
+                mode=mode,
+                label_smoothing=label_smoothing,
+                num_classes=num_classes)
+
+    def extract_feat(self, img):
+        """Function to extract features from backbone.
+
+        Args:
+            img (Tensor): Input images of shape (N, C, H, W).
+                Typically these should be mean centered and std scaled.
+
+        Returns:
+            tuple[Tensor]: backbone outputs.
+        """
+        x = self.backbone(img)
+        return x
+
+    def forward_train(self, img, label, **kwargs):
+        """Forward computation during training.
+
+        Args:
+            img (Tensor): Input images of shape (N, C, H, W).
+                Typically these should be mean centered and std scaled.
+            label (Tensor): Ground-truth labels.
+            kwargs: Any keyword arguments to be used to forward.
+
+        Returns:
+            dict[str, Tensor]: A dictionary of loss components.
+        """
+        if self.finetune:
+            img, label = self.mix_up(img, label)
```

Comment:
The usage of mix-up should not be related to self.finetune. Use a specific flag (e.g., use_mixup) to control whether to use mix_up.

Reply:
Thank you for your valuable comments. Your suggestion is reasonable, and I will fix it.